core/src/main/java/hudson/model/Run.java

```diff
@@ -953,17 +953,32 @@ protected void dropLinks() {
      * @since 1.383
      */  
     public @Nonnull List<RunT> getPreviousBuildsOverThreshold(int numberOfBuilds, @Nonnull Result threshold) {
-        List<RunT> builds = new ArrayList<>(numberOfBuilds);
-        
         RunT r = getPreviousBuild();
+        return getBuildsOverThreshold(r, numberOfBuilds, threshold);
+    }
+
+    /**
+     * Returns the last {@code numberOfBuilds} builds with a build result ≥ {@code threshold}.
+     *
+     * @param r the run to search
+     * @param numberOfBuilds the desired number of builds
+     * @param threshold the build result threshold
+     * @return a list with the builds (youngest build first).
+     *   May be smaller than 'numberOfBuilds' or even empty
+     *   if not enough builds satisfying the threshold have been found. Never null.
+     * @since TODO
+     */
+    protected @Nonnull List<RunT> getBuildsOverThreshold(RunT r, int numberOfBuilds, @Nonnull Result threshold) {
```

Comment:
```suggestion
    protected @Nonnull List<RunT> getBuildsOverThreshold(@CheckForNull RunT r, int numberOfBuilds, @Nonnull Result threshold) {
```

Reply:
I'm going to undo this per @res0nance below. Please wait until people are happy w/ the alternate approach.